crates/rpc/tests/rpc.rs

```diff
@@ -0,0 +1,477 @@
+// These tests need Beerus to run in the background, hence why they're hidden behind the following feature.
+#![cfg(feature = "integration-tests")]
+
+use cached::Cached;
+use std::ops::Deref;
+
+use cached::SizedCache;
+use reqwest::Url;
+use starknet::{
+    core::types::{
+        BlockId, BlockWithTxHashes, BlockWithTxs, DeclareTransaction,
+        DeployAccountTransaction, FieldElement, InvokeTransaction,
+        MaybePendingBlockWithTxHashes, MaybePendingBlockWithTxs,
+        MaybePendingTransactionReceipt, Transaction, TransactionReceipt,
+    },
+    providers::{
+        jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError,
+    },
+};
+
+struct TestContext<T> {
+    client: TestClient,
+    block: BlockWithTxs,
+    block_id: BlockId,
+    extracted_value: T,
+}
+
+/// Creates a `TestContext` with the latest block.
+async fn latest_block_context() -> TestContext<()> {
+    context(|_block| Some(())).await
```

Comment:
If I understand correctly, what `latest_block_context` is doing is pulling latest block from the RPC and then drops it to return `()` as the extracted value. So the RPC call can be dropped, right?

Won't having two fixtures be simpler (just a question, not a suggestion)? Then each test case can use whichever is necessary and no redundant RPC calls are made:
- `with_client` that returns `TestClient`
- `with_client_and_block` that returns `(TestClient, BlockWithTxs, BlockId)`

Reply:
I separated the creation of the RPC client from the test context and got rid of the `test_block_number` scenario as it's redundant indeed.
[Relevant push](https://github.com/eigerco/beerus/compare/66420d6accb4502a53cdd67fab9ea9a0822c81ed..65fcee33d079b56d8e3bc88a1d51a219c05f6f1d).